### ISSUE TITLE
Declination column missing when writing tables with SkyCoord columns to ECSV

### ISSUE THREAD

**Issue by lpsinger** [astropy/astropy#16998]
### Description

If you create a new SkyCoord instance from an existing one, passing the `representation_type=` kwarg to the constructor, add it to a table, and write the table to ECSV, columns are missing.

For example, this code:

```python
coords = SkyCoord(np.zeros(5), np.ones(5), np.zeros(5), representation_type=CartesianRepresentation)
coords = SkyCoord(coords, representation_type=UnitSphericalRepresentation)
Table({"coords": coords}).write(sys.stdout, format="ascii.ecsv")
```

results in the following table output, missing the declination column:

```
# %ECSV 1.0
# ---
# datatype:
# - {name: coords.ra, unit: deg, datatype: float64}
# meta: !!omap
# - __serialized_columns__:
#     coords:
#       __class__: astropy.coordinates.sky_coordinate.SkyCoord
#       frame: icrs
#       ra: !astropy.table.SerializedColumn
#         __class__: astropy.coordinates.angles.core.Longitude
#         unit: &id001 !astropy.units.Unit {unit: deg}
#         value: !astropy.table.SerializedColumn {name: coords.ra}
#         wrap_angle: !astropy.coordinates.Angle
#           unit: *id001
#           value: 360.0
#       representation_type: unitspherical
# schema: astropy-2.0
coords.ra
90.0
90.0
90.0
90.0
90.0
```

### Expected behavior

Each of the three examples in the code below should print:

```
# %ECSV 1.0
# ---
# datatype:
# - {name: coords.ra, unit: deg, datatype: float64}
# - {name: coords.dec, unit: deg, datatype: float64}
# meta: !!omap
# - __serialized_columns__:
#     coords:
#       __class__: astropy.coordinates.sky_coordinate.SkyCoord
#       dec: !astropy.table.SerializedColumn
#         __class__: astropy.coordinates.angles.core.Latitude
#         unit: &id001 !astropy.units.Unit {unit: deg}
#         value: !astropy.table.SerializedColumn {name: coords.dec}
#       frame: icrs
#       ra: !astropy.table.SerializedColumn
#         __class__: astropy.coordinates.angles.core.Longitude
#         unit: *id001
#         value: !astropy.table.SerializedColumn {name: coords.ra}
#         wrap_angle: !astropy.coordinates.Angle
#           unit: *id001
#           value: 360.0
#       representation_type: spherical
# schema: astropy-2.0
coords.ra coords.dec
90.0 0.0
90.0 0.0
90.0 0.0
90.0 0.0
90.0 0.0
```

However, only the first two tables print that. The last table is missing the declination column.

### How to Reproduce

```python
from astropy import units as u
from astropy.coordinates import CartesianRepresentation, UnitSphericalRepresentation, SkyCoord
import numpy as np
from astropy.table import Table
import sys

coords = SkyCoord(np.full(5, 90), np.zeros(5), unit=u.deg)
Table({"coords": coords}).write(sys.stdout, format="ascii.ecsv")

coords = SkyCoord(np.zeros(5), np.ones(5), np.zeros(5), representation_type=CartesianRepresentation)
coords.representation_type = UnitSphericalRepresentation
Table({"coords": coords}).write(sys.stdout, format="ascii.ecsv")

coords = SkyCoord(np.zeros(5), np.ones(5), np.zeros(5), representation_type=CartesianRepresentation)
coords = SkyCoord(coords, representation_type=UnitSphericalRepresentation)
Table({"coords": coords}).write(sys.stdout, format="ascii.ecsv")
```

### Versions

```
Python 3.11.10 (main, Sep  7 2024, 05:47:53) [Clang 15.0.0 (clang-1500.1.0.2.5)]
astropy 6.1.3
Numpy 2.1.1
pyerfa 2.0.1.4
Scipy 1.14.1
Matplotlib 3.9.2
```

**Comment by eerovaher** [comment]
I don't have a solution, but it very much looks like the bug is in `coordinates`, not in `io`:
```python
>>> from astropy.coordinates import SkyCoord
>>> sc = SkyCoord(0, 0, unit="deg", representation_type="unitspherical")
>>> sc.info._represent_as_dict()
{'ra': <Longitude 0. deg>, 'representation_type': 'unitspherical', 'frame': 'icrs'}
```

**Comment by mhvk** [comment]
Thanks for reporting, and thanks, @eerovaher, for finding the underlying problem - PR with fix forthcoming.

**Comment by mhvk** [comment]
Took longer than I thought since there were similar problems with PM and RV as well... Plus another bug... Anyway, fix in #16999.
